vignettes/ebola.Rmd

```diff
@@ -0,0 +1,615 @@
+
+```{r epred-base, fig.cap="(ref:epred-base)"}
+epred_draws |>
+  ggplot(aes(x = .epred)) +
+  tidybayes::stat_slab() +
+  labs(x = "Posterior expectation of the delay", y = "") +
+  theme_minimal()
+```
+
+Next, for the sex stratified model, we illustrate the higher mean of men as compared with women (Figure \@ref(fig:epred-sex)):
+
+```{r}
+# Probably want functionality for users not to have to do this themselves
+(newdata_sex <- obs_prep |>
+   as.data.frame() |>
+   modelr::data_grid(sex) |>
+   dplyr::mutate(obs_t = NA, pwindow_upr = NA, swindow_upr = NA))
+
+epred_draws_sex <- tidybayes::add_epred_draws(newdata_sex, fit_sex, dpar = TRUE)
+```
+
+(ref:epred-sex) The fitted posterior expectation of the delay distribution for a model with sex stratification.
+
+```{r epred-sex, fig.cap="(ref:epred-sex)"}
+epred_draws_sex |>
+  dplyr::mutate(
+    sex = dplyr::case_when(
+      sex == "F" ~ "Female",
+      sex == "M" ~ "Male"
+    )
+  ) |>
+  ggplot(aes(x = .epred, y = sex)) +
+  tidybayes::stat_halfeye() +
+  labs(x = "Posterior expectation of the delay", y = "") +
+  theme_minimal()
+```
+
+Finally, the sex-district stratified model is shown in Figure \@ref(fig:epred-sex).
+Note that these results are questionable, and suggest that further would should be done refining the prior distributions for the model.
+Part of the issue is that some sex-district strata have quite limited data.
+
+```{r}
+epred_draws_sex_district <- obs_prep |>
+  as.data.frame() |>
+  modelr::data_grid(sex, district) |>
+  dplyr::mutate(obs_t = NA, pwindow_upr = NA, swindow_upr = NA) |>
+  add_epred_draws(fit_sex_district, dpar = TRUE)
+```
+
+(ref:epred-sex-district) The fitted posterior expectation of the delay distribution for a model with sex-district stratification.
+
+```{r epred-sex-district, fig.cap="(ref:epred-sex-district)"}
+epred_draws_sex_district |>
+  dplyr::mutate(
+    sex = dplyr::case_when(
+      sex == "F" ~ "Female",
+      sex == "M" ~ "Male"
+    )
+  ) |>
+  ggplot(aes(x = .epred, y = district)) +
+  tidybayes::stat_pointinterval() +
+  facet_grid(. ~ sex) +
+  labs(x = "Posterior expectation of the delay", y = "") +
+  theme_minimal()
+```
+
+## Linear predictor posteriors
+
+The `tidybayes` package also allows users to generate draws of the linear predictors for all distributional parameters using `add_linpred_draws`.
+For example, for the `mu` parameter in the sex-district stratified model:
+
+```{r}
+linpred_draws_sex_district <- obs_prep |>
+  as.data.frame() |>
+  modelr::data_grid(sex, district) |>
+  dplyr::mutate(obs_t = NA, pwindow_upr = NA, swindow_upr = NA) |>
+  add_linpred_draws(fit_sex_district, dpar = TRUE)
+```
+
+(ref:linpred-sex-district) The posterior distribution of the linear predictor of `mu` parameter within the sex-district stratified model.
+
+```{r linpred-sex-district, fig.cap="(ref:linpred-sex-district)"}
+linpred_draws_sex_district |>
+  dplyr::mutate(
+    sex = dplyr::case_when(
```

Comment:
why do the male predictions has so little uncertainty?

Reply:
Failure with Laplace inference. Was fixed by using HMC https://github.com/epinowcast/epidist/issues/289